File: src/modifiers/buttonRepeatBehavior.ts
import { createModifier } from './createModifier';

/// Whether buttons repeatedly trigger their action on a press-and-hold.
export type ButtonRepeatBehaviorValue = 'automatic' | 'enabled' | 'disabled';

/// SwiftUI `.buttonRepeatBehavior(_:)`. Controls whether descendant buttons
/// fire their action repeatedly while held down.
///
/// Requires watchOS 10+. On earlier systems this is a no-op (the view is
/// returned unchanged).
export function buttonRepeatBehavior(
  behavior: ButtonRepeatBehaviorValue = 'automatic'
) {
  return createModifier('buttonRepeatBehavior', { behavior });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/ButtonRepeatBehavior.swift
import SwiftUI

/// SwiftUI `.buttonRepeatBehavior(_:)`. Controls whether descendant buttons
/// fire their action repeatedly while held down.
///
/// Requires watchOS 10; on earlier systems the view is returned unchanged.
enum RNWButtonRepeatBehaviorModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("buttonRepeatBehavior") { view, params, _ in
            guard #available(watchOS 10.0, *) else { return view }
            switch params.string("behavior") {
            case "enabled":  return AnyView(view.buttonRepeatBehavior(.enabled))
            case "disabled": return AnyView(view.buttonRepeatBehavior(.disabled))
            default:         return AnyView(view.buttonRepeatBehavior(.automatic))
            }
        }
    }
}
